src/PreCICE.jl

```diff
@@ -64,227 +85,1031 @@ function createSolverInterfaceWithCommunicator(participantName::String,
+edge IDs, since it needs to check, whether an edge is created already or not.
+
+# Arguments
+- `meshID::Integer`: ID of the mesh to add the Quad to.
+- `firstVertexID::Integer`: ID of the first edge of the Quad.
+- `secondVertexID::Integer`: ID of the second edge of the Quad.
+- `thirdEdgeID::Integer`: ID of the third edge of the Quad.
+- `fourthEdgeID::Integer`: ID of the fourth edge of the Quad.
+
+Notes
+
+Previous calls:
+ - Edges with `firstVertexID`, `secondEdgeID`, `thirdVertexID`, and `fourthEdgeID` were added
+    to the mesh with the ID `mesh_id`
+"""
 function setMeshQuadWithEdges(meshID::Integer, firstEdgeID::Integer, secondEdgeID::Integer, thirdEdgeID::Integer)
     ccall((:precicec_setMeshQuadWithEdges, "libprecice"), Cvoid, (Cint, Cint, Cint, Cint, Cint), meshID, firstEdgeID, secondEdgeID, thirdEdgeID, fourthEdgeID)
 end
 
 
+# TODO is the form of the vector correct? or can this be passed as a matrix instead of a vector?
+@doc """
+
+    writeBlockVectorData(dataID::Integer, size::Integer, valueIndices::AbstractArray{Cint}, values::AbstractArray{Float64})
+
+Write vector data values given as block. This function writes values of specified vertices to a `dataID`.
+Values are provided as a block of continuous memory. Values are stored in a Matrix [N x D] where N = number
+of vertices and D = dimensions of geometry 
+
+The block must contain the vector values in the following form: 
+
+values = (d0x, d0y, d0z, d1x, d1y, d1z, ...., dnx, dny, dnz), where n is the number of vector values. In 2D, the z-components are removed.
+
+# Arguments
+- `dataID::Integer`: ID of the data to be written.
+- `size::Integer`: Number n of vertices. 
+- `valueIndices::AbstractArray{Cint}`: Indices of the vertices. 
+- `values::AbstractArray{Float64}`: Values of the data to be written.
+
+# Notes
+
+Previous calls:
+ - count of available elements at `values` matches the configured `dimension` * `size`
+ - count of available elements at `vertex_ids` matches the given size
+ - [`initialize`](@ref) has been called
+
+Examples
+
+Write block vector data for a 2D problem with 5 vertices:
+```julia
+data_id = 1
+vertex_ids = [1, 2, 3, 4, 5]
+values = [v1_x, v1_y; v2_x, v2_y; v3_x, v3_y; v4_x, v4_y; v5_x, v5_y])
+writeBlockVectorData(data_id, vertex_ids, values)
+```
+Write block vector data for a 3D (D=3) problem with 5 (N=5) vertices:
+```julia
+data_id = 1
+vertex_ids = [1, 2, 3, 4, 5]
+values = [v1_x, v1_y, v1_z; v2_x, v2_y, v2_z; v3_x, v3_y, v3_z; v4_x, v4_y, v4_z; v5_x, v5_y, v5_z]
+writeBlockVectorData(data_id, vertex_ids, values)
+```
+"""
 function writeBlockVectorData(dataID::Integer, size::Integer, valueIndices::AbstractArray{Cint}, values::AbstractArray{Float64})
     ccall((:precicec_writeBlockVectorData, "libprecice"), Cvoid, (Cint, Cint, Ref{Cint}, Ref{Cdouble}), dataID, size, valueIndices, values)
 end
 
 
+# TODO Are they provided as a block of continuous memory?
```

Comment:
I dont really understand what you mean here, can you please elaborate?

Reply:
In the python bindings it said that the Values would be provided as a block of continuous memory. Is this also true in this case or would we be on the safe side not including this part?